minion/ovsdb/ovsdb.go

```
@@ -67,779 +100,724 @@ type AclCore struct {
+	}
+	iface.Name = name
+
+	ifaceType, ok := row["type"].(string)
+	if !ok {
+		return iface, errors.New("interface missing type")
+	}
+	iface.Type = ifaceType
+
+	optRow, ok := row["options"]
+	if !ok {
+		return iface, errors.New("missing Interface key: options")
+	}
+	options, err := ovsStringMapToMap(optRow)
 	if err != nil {
-		return fmt.Errorf("transaction error updating interface %s: %s",
-			name, err)
+		return iface, err
 	}
-	return errorCheck(results, 1, 1)
+
+	extRow, ok := row["external_ids"]
+	if !ok {
+		return iface, errors.New("missing Interface key: external_ids")
+	}
+	externalIDs, err := ovsStringMapToMap(extRow)
+	if err != nil {
+		return iface, err
+	}
+
+	// The following map keys could be missing without breaking the Schema in the
+	// Interface table.
+	if peer, ok := options["peer"]; ok {
+		iface.Peer = peer
+	} else {
+		log.Debug("interface missing peer.")
+	}
+
+	if amac, ok := externalIDs["attached-mac"]; ok {
+		iface.AttachedMAC = amac
+	} else {
+		log.Debug("interface missing attached-mac.")
+	}
+
+	if id, ok := externalIDs["iface-id"]; ok {
+		iface.IfaceID = id
+	} else {
+		log.Debug("interface missing iface-id.")
+	}
+
+	return iface, nil
 }
 
-// SetOFInterfacePeer sets the peer of the interface
-func (ovsdb Client) SetOFInterfacePeer(name, peer string) error {
-	mut := newMutation("options", "insert", map[string]string{"peer": peer})
-	if err := ovsdb.addToOFInterface(name, mut); err != nil {
-		return fmt.Errorf("error setting interface %s to peer %s: %s",
-			name, peer, err)
+// This does not cover all cases, they should just be added as needed
+func newMutation(column, mutator string, value interface{}) mutation {
+	switch typedValue := value.(type) {
+	case ovs.UUID:
+		uuidSlice := []ovs.UUID{typedValue}
+		mutateValue, err := ovs.NewOvsSet(uuidSlice)
+		if err != nil {
+			// An error can only occur if the input is not a slice
+			panic("newMutation(): an impossible error occurred")
```

This was actually old code, just changed place. So yeah, there is no way it would have error, and other people's code just ignore the error as well, so I guess we could just ignore it too.